### ISSUE TITLE
"Upgrade: Temporary Etcd3" step fails when using DNS

### ISSUE THREAD

**Issue by hreidar** [apprenda/kismatic#534]
Hi, I'm getting an error in one of the upgradings steps. I'm trying to upgrade from v.1.2.2 to v.1.3.2.

I have already upgraded a dev server successfully but the difference there I'm using "update_hosts_files: true"

If I run etcdctl on this endpoint using IP address it returns cluster is healthy ...

```
root@kube-etcd1:~# etcdctl --endpoint='https://127.0.0.1:6666' cluster-health
member 458f2e6680fe97b3 is healthy: got healthy result from https://83.173.15.171:6666
member 87545e1a7725a1e1 is healthy: got healthy result from https://83.173.15.170:6666
member d880825442e4eeae is healthy: got healthy result from https://83.173.15.172:6666
cluster is healthy
root@kube-etcd1:~# etcdctl --endpoint='https://10.10.10.170:6666' cluster-health          
member 458f2e6680fe97b3 is healthy: got healthy result from https://83.173.15.171:6666
member 87545e1a7725a1e1 is healthy: got healthy result from https://83.173.15.170:6666
member d880825442e4eeae is healthy: got healthy result from https://83.173.15.172:6666
cluster is healthy
```
... but not if I use DNS.
```
root@kube-etcd1:~# etcdctl --endpoint='https://kube-etcd1:6666' cluster-health         
cluster may be unhealthy: failed to list members
Error:  client: etcd cluster is unavailable or misconfigured
error #0: dial tcp 127.0.1.1:6666: getsockopt: connection refused
```

Here is the ERROR from kismatic/ansible:
```
Upgrade: kube-etcd1 to Temporary Etcd3==============================================
Configure Kismatic Package Repos                                                [OK]
Temporarily Install Transition Etcd 3.x version                                 [ERROR]
- Task: get etcd server version
  kube-etcd1:                                                                   [ERROR]
---- STDERR ----
  %!T(MISSING)otal    %!R(MISSING)eceived %!X(MISSING)ferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to kube-etcd1 port 6666: Connection refused
---------------
Failed to upgrade etcd2 nodes: error upgrading node "kube-etcd1": error running playbook: error running ansible: exit status 2
```
Here is my kismatic-cluster.yaml
```
cluster:
  name: kube-cluster1
  admin_password: xxxxxxxxxx
  allow_package_installation: true
  disconnected_installation: false
  networking:
    type: routed
    pod_cidr_block: 172.16.0.0/16
    service_cidr_block: 172.17.0.0/16
    policy_enabled: false
    update_hosts_files: false
  certificates:
    expiry: 43800h
  ssh:
    user: kismaticuser
    ssh_key: /opt/kismatic.1.3.2/kismaticuser.key
    ssh_port: 22
docker_registry:
  setup_internal: true
  address: ""
  port: 8443
  CA: ""
etcd:
  expected_count: 3
  nodes:
  - host: kube-etcd1
    ip: 10.10.10.170
    internalip: ""
  - host: kube-etcd2
    ip: 10.10.10.171
    internalip: ""
  - host: kube-etcd3
    ip: 10.10.10.172
    internalip: ""
master:
  expected_count: 2
  load_balanced_fqdn: 10.10.10.173
  load_balanced_short_name: 10.10.10.173
  nodes:
  - host: kube-master1
    ip: 10.10.10.173
    internalip: ""
  - host: kube-master2
    ip: 10.10.10.174
    internalip: ""
worker:
  expected_count: 2
  nodes:
  - host: kube-worker1
    ip: 10.10.10.175
    internalip: ""
  - host: kube-worker2
    ip: 10.10.10.176
    internalip: ""
ingress:
  expected_count: 0
  nodes: []
storage:
  expected_count: 0
  nodes: []
nfs:
  nfs_volume: []
```

**Comment by dkoshkin** [comment]
Hi @hreidar are you confident that the DNS is working correctly? Are you able to `ping kube-etcd1` from the other etcd nodes?

**Comment by alexbrand** [comment]
@hreidar This looks like a misconfiguration to me. From the output you posted, it looks like `kube-etcd1` resolves to `127.0.1.1`. I would guess that is an invalid IP, which is why the connection fails.

```
root@kube-etcd1:~# etcdctl --endpoint='https://kube-etcd1:6666' cluster-health         
cluster may be unhealthy: failed to list members
Error:  client: etcd cluster is unavailable or misconfigured
error #0: dial tcp 127.0.1.1:6666: getsockopt: connection refused
```

I am going to close this. Feel free to re-open if you run into further issues. We also have a Slack channel where we can help you troubleshoot. The link to join the Slack team is visible in our main readme.

**Comment by hreidar** [comment]
Hi, you are absolutely right, I did not notice that, thanks.
